app/webpacker/components/RegistrationsV2/Registrations/Competitors.jsx

```diff
@@ -101,164 +150,68 @@ export default function Competitors({
-      </Table>
-    </>
-  );
-}
-
-function CompetitorsHeader({
-  eventIds,
-  sortedColumn,
-  sortedDirection,
-  onSortableColumnClick,
-  onEventColumnClick,
-}) {
-  return (
-    <Table.Header>
-      <Table.Row>
-        <Table.HeaderCell
-          sorted={sortedColumn === 'name' ? sortedDirection : undefined}
-          onClick={() => onSortableColumnClick('name')}
-        >
-          {I18n.t('activerecord.attributes.registration.name')}
-        </Table.HeaderCell>
-        <Table.HeaderCell
-          sorted={sortedColumn === 'country' ? sortedDirection : undefined}
-          onClick={() => onSortableColumnClick('country')}
-        >
-          {I18n.t('activerecord.attributes.user.country_iso2')}
-        </Table.HeaderCell>
-        {eventIds.map((id) => (
-          <Table.HeaderCell
-            textAlign="center"
-            key={`registration-table-header-${id}`}
-            onClick={() => onEventColumnClick(id)}
-          >
-            <EventIcon id={id} size="1em" className="selected" />
-          </Table.HeaderCell>
-        ))}
-        <Table.HeaderCell
-          textAlign="center"
-          sorted={sortedColumn === 'total' ? sortedDirection : undefined}
-          onClick={() => onSortableColumnClick('total')}
-        >
-          {I18n.t('registrations.list.total')}
-        </Table.HeaderCell>
-      </Table.Row>
-    </Table.Header>
-  );
-}
-
-function CompetitorsBody({
-  registrations,
-  eventIds,
-  userId,
-  userRowRef,
-}) {
-  return (
-    <Table.Body>
-      {registrations.length > 0 ? (
-        registrations.map((registration) => {
-          const isUser = registration.user_id === userId;
-          return (
-            <Table.Row
-              key={`registration-table-row-${registration.user.id}`}
-              active={isUser}
-            >
-              <Table.Cell>
-                <div ref={isUser ? userRowRef : undefined}>
-                  {registration.user.wca_id ? (
-                    <a
-                      href={personUrl(registration.user.wca_id)}
+        <Table.Header>
+          {table.getHeaderGroups().map((headerGroup) => (
+            <Table.Row key={headerGroup.id}>
+              {headerGroup.headers.map((header) => (
+                <Table.HeaderCell
+                  key={header.id}
+                  sorted={expandSortTerm(header.column.getIsSorted())}
+                  onClick={
+                    header.column.getCanSort()
+                      ? header.column.getToggleSortingHandler()
+                      : () => onEventClick(header.column.id)
+                  }
+                  textAlign={header.column.columnDef.meta?.style?.textAlign}
+                >
+                  {flexRender(header.column.columnDef.header, header.getContext())}
+                </Table.HeaderCell>
+              ))}
+            </Table.Row>
+          ))}
+        </Table.Header>
+        <Table.Body>
+          {table.getSortedRowModel().rows.length > 0 ? (
```

Comment:
Can you return early if no columns are found?

Reply:
Do you mean rows? Returning early would omit the header/footer. That could work, I guess. I prefer having the header/footer, and I think many of our other tables do that, including as the psych sheet.